xarray/backends/api.py

```diff
@@ -1699,21 +1701,26 @@ def to_zarr(
     store = _normalize_path(store)
     chunk_store = _normalize_path(chunk_store)
 
+    kwargs = {}
     if storage_options is None:
         mapper = store
         chunk_mapper = chunk_store
     else:
-        from fsspec import get_mapper
-
         if not isinstance(store, str):
             raise ValueError(
                 f"store must be a string to use storage_options. Got {type(store)}"
             )
-        mapper = get_mapper(store, **storage_options)
-        if chunk_store is not None:
-            chunk_mapper = get_mapper(chunk_store, **storage_options)
+
+        if _zarr_v3():
+            kwargs["storage_options"] = storage_options
```

Comment:
This branch needs to define mapper` and `chunk_mapper` as `store` and `chunk_store`.

Reply:
I discovered this while doing some manual tests for the issue at https://github.com/pydata/xarray/pull/9552#issuecomment-2412109812. Still tracking down one more issue, but I'm reasonably confident it's on the Zarr side.